### ISSUE TITLE
ag not printing empty files with -ul

### ISSUE THREAD

**Issue by HaleTom** [ggreer/the_silver_searcher#956]
From the man page:  (`ag version 0.28.0`)

> ```
>    -l --files-with-matches
>           Only print the names of files containing matches, not the matching lines. An empty query will print all files that would be searched.
> 
>    -u --unrestricted
>           Search all files. This ignores .agignore, .gitignore, etc. It searches binary and hidden files as well.
> ```

So I'd expect `ag -lu` to list all files.

However:

```
+ravi@boxy:~$ mkdir new && cd new
+ravi@boxy:~/new$ echo stuff > non-empty && touch empty
+ravi@boxy:~/new$ ag -ul
non-empty
+ravi@boxy:~/new$
```

**Comment by HaleTom** [comment]
There has been some discussion (including code) at:
[How do I force ag (the sliver searcher) to list empty files](http://unix.stackexchange.com/questions/303326/how-do-i-force-ag-the-sliver-searcher-to-list-empty-files)

**Comment by ggreer** [comment]
Fixed in #957. Thanks!
